tasks/main.yml

```diff
@@ -86,14 +86,27 @@
   template: src='pxe_nodes.json.j2' dest='/var/www/provision/nodes/pxe_nodes.json' backup=yes
   tags: pxe_data
 
-- name: create_kickstart_group_files
+- name: create_kickstart_group_files for legacy nodes
   template: src=kickstart.j2 dest="/var/www/html/kickstart/{{item}}.ks" validate='/usr/bin/ksvalidator %s' backup=yes
   tags: dhcp_kickstart_config
   with_items: "{{ groups|sort }}"
-  when:
-    - item not in dhcp_kickstart_skip_these_groups
-    - groups[item][0] is defined
-    - hostvars[groups[item][0]]['os_disks'] is defined
+  when: |
+    item not in dhcp_kickstart_skip_these_groups and
+    groups[item][0] is defined and
+    hostvars[groups[item][0]]['os_disks'] is defined and
+    ansible_os_family == "RedHat" and
+    ansible_distribution_major_version <= "7"
```

Comment:
This ansible_distribution_major_version is checked when the playbook is run, and for nodes currently installed with CentOS 7, this will be 7. This means for nodes currently running CentOS 7, a kickstart file for CentOS 7 will be generated.

Reply:
I suggest adding a group variable declaring which CentOS version should be installed on these nodes, rather than checking what's running now and going with the same.

Reply:
Should be resolved now.